File: ios/MyTwitter/AppConfig.swift
import Foundation

enum AppConfig {
    static var siteURL: String {
        let raw = (Bundle.main.object(forInfoDictionaryKey: "SITE_URL") as? String)?
            .trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        let cleaned = raw.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
        if cleaned.isEmpty || cleaned.contains("YOUR_PROJECT_ID") {
            return "https://YOUR_PROJECT_ID.web.app"
        }
        return cleaned
    }

    /// Empty / placeholder / unresolved xcconfig ⇒ Sentry stays off (safe for forks).
    static var sentryDSN: String {
        let raw = (Bundle.main.object(forInfoDictionaryKey: "SENTRY_DSN") as? String)?
            .trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
        if raw.isEmpty { return "" }
        if raw.contains("$(") { return "" }
        if raw.localizedCaseInsensitiveContains("YOUR_") { return "" }
        if !(raw.hasPrefix("http://") || raw.hasPrefix("https://")) { return "" }
        return raw
    }

    static var versionName: String {
        Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "0.1.0"
    }

    static var versionCode: Int {
        Int(Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? "1") ?? 1
    }

    static var siteHost: String? {
        URL(string: siteURL)?.host?.lowercased()
    }
}

File: android/app/src/main/java/org/evergreenlabs/mytwitter/AppConfig.kt
package org.evergreenlabs.mytwitter

import android.net.Uri

object AppConfig {
    val siteUrl: String
        get() {
            val raw = BuildConfig.SITE_URL.trim().trimEnd('/')
            if (raw.isEmpty() || raw.contains("YOUR_PROJECT_ID")) {
                return "https://YOUR_PROJECT_ID.web.app"
            }
            return raw
        }

    val sentryDsn: String
        get() {
            val raw = BuildConfig.SENTRY_DSN.trim()
            if (raw.isEmpty()) return ""
            if (raw.contains("YOUR_", ignoreCase = true)) return ""
            if (!raw.startsWith("http://") && !raw.startsWith("https://")) return ""
            return raw
        }

    val versionName: String
        get() = BuildConfig.VERSION_NAME

    val versionCode: Int
        get() = BuildConfig.VERSION_CODE

    val siteHost: String?
        get() = Uri.parse(siteUrl).host?.lowercase()
}
